### DISCUSSION TITLE
Convert multiple .gif to .png with each output in seperate folder

### DISCUSSION THREAD

**Discussion by Dooey123** [ImageMagick/ImageMagick#7757] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows 11

### Description

I am able to convert a single .gif to individual .png frame images using **convert -coalesce "path\*.gif" "path\Frame%%0d.png"** which works great. 

Is there a way convert multiple .gif files in the source path where each set of output frames exports to a new folder in the destination path?

**Comment by fmw42** [comment]
Not sure I understand the question. Can you give a set of examples of what you want for the paths, outputs and inputs. Most likely you would have to create multiple command lines. What are your folder names? Are you asking  for each output to go to different folders or the same folder? Are you gifs, single frame or multi-frame?

You can process multiple gifs to go to the same folder using magick mogrify.

Please note that on IM 7, use magick, not convert. And read the input right after magick. Put -coalesce after the input.

**Reply by Dooey123** [reply]
This is my current working command that I run from batch file: 

`"C:\Program Files\ImageMagick-7.1.1-Q16-HDRI\magick.exe" convert -coalesce "C:\Users\dpsro\Desktop\*.gif" "C:\Users\dpsro\Desktop\Frames\Frame%%0d.png"`

but instead of running this command on a single gif one at a time and outputting to a single folder - to save time I would like to be able to run this on multiple gif files and for each file the output images are sent to a new folder. 

For example gif1 outputs to folder "Frames1", gif2 outputs to folder "Frames2" etc. for each gif found on the desktop. I'd like the batch script to create each folder automatically for each gif it can find.

**Reply by fmw42** [reply]
Imagemagick does not create new folders. They must exist already.  You should create FOR loop over existing directories and use magick not convert and put -coalesce after reading the input gifs. Or loop over magick mogrify for all gifs in a given directory and write to some existing directory.
